app/services/api/unfunded_mentors/query.rb

```diff
@@ -0,0 +1,78 @@
+module API::UnfundedMentors
+  class Query
+    include Queries::FilterIgnorable
+
+    attr_reader :scope, :lead_provider_id
+
+    def initialize(
+      lead_provider_id: :ignore,
+      updated_since: :ignore,
+      sort: { created_at: :asc }
+    )
+      @lead_provider_id = lead_provider_id
+      @scope = Teacher.distinct
+
+      where_lead_provider_is(lead_provider_id)
+      where_updated_since(updated_since)
+      set_sort_by(sort)
+    end
+
+    def unfunded_mentors
+      preload_associations(block_given? ? yield(scope) : scope)
+    end
+
+    def unfunded_mentor_by_api_id(api_id)
+      return preload_associations(scope).find_by!(api_id:) if api_id.present?
+
+      fail(ArgumentError, "api_id needed")
+    end
+
+    def unfunded_mentor_by_id(id)
+      return preload_associations(scope).find(id) if id.present?
+
+      fail(ArgumentError, "id needed")
+    end
+
+  private
+
+    def preload_associations(results)
+      # Joining with `mentor_at_school_periods`
+      # In case we don't filter by lead provider
+      # We still want to return only mentors
+      results
+        .strict_loading
+        .joins(:mentor_at_school_periods)
+        .includes(
+          :latest_mentor_at_school_period
+        )
+    end
+
+    def where_lead_provider_is(lead_provider_id)
```

Comment:
Did you look at doing it in a single query? I'm wondering how performant this is; can we maybe test it on the parity check environment? if its fast as-is then great, but if not we may need to look into combining into a single query.

Reply:
Ok happy to try it as-is as I think it would be harder to follow in a single query. The parity check will give us a comparison with ECF as well 👍